mythril/laser/smt/bitvec.py

```diff
@@ -160,14 +181,282 @@ def __ne__(self, other: Union[int, "BitVec"]) -> Bool:  # type: ignore
+
+    def __lt__(self, other: "BitVec") -> Bool:
+        """Create a signed less than expression.
+
+        :param other:
+        :return:
+        """
+        # Is there some hack for these comparisons?
+
+        if not isinstance(other, BitVec):
+            other = BitVec(z3.BitVecVal(other, self.size()))
+
+        union = self.annotations + other.annotations
+
+        if not self.symbolic and not other.symbolic:
+            return Bool(z3.BoolVal(self.value < other.value), annotations=union)
+
+        if (
+            not isinstance(other, BitVecFunc)
+            or not self.name
+            or not self.input
+            or not self.name == other.name
+        ):
+            return Bool(z3.BoolVal(False), annotations=union)
+
+        return And(
+            Bool(cast(z3.BoolRef, self.raw < other.raw), annotations=union),
+            self.input != other.input,
+        )
+
+    def __gt__(self, other: "BitVec") -> Bool:
+        """Create a signed greater than expression.
+
+        :param other:
+        :return:
+        """
+        # Is there some hack for these comparisons?
+
+        if not isinstance(other, BitVec):
+            other = BitVec(z3.BitVecVal(other, self.size()))
+
+        union = self.annotations + other.annotations
+
+        if not self.symbolic and not other.symbolic:
+            return Bool(z3.BoolVal(self.value > other.value), annotations=union)
+
+        if (
+            not isinstance(other, BitVecFunc)
+            or not self.name
+            or not self.input
+            or not self.name == other.name
+        ):
+            return Bool(z3.BoolVal(False), annotations=union)
+
+        return And(
+            Bool(cast(z3.BoolRef, self.raw > other.raw), annotations=union),
+            self.input != other.input,
+        )
+
+    # MYPY: fix complains about overriding __eq__
+    def __eq__(self, other: Union[int, "BitVec"]) -> Bool:  # type: ignore
+        """Create an equality expression.
+
+        :param other:
+        :return:
+        """
+        if not isinstance(other, BitVec):
+            other = BitVec(z3.BitVecVal(other, self.size()))
+
+        union = self.annotations + other.annotations
+
+        if not self.symbolic and not other.symbolic:
+            return Bool(z3.BoolVal(self.value == other.value), annotations=union)
+
+        if (
+            not isinstance(other, BitVecFunc)
+            or not self.name
+            or not self.input
+            or not self.name == other.name
+        ):
+            return Bool(z3.BoolVal(True), annotations=union)
+
+        # MYPY: fix complaints due to z3 overriding __eq__
+        return And(
+            Bool(cast(z3.BoolRef, self.raw == other.raw), annotations=union),
+            self.input == other.input,
+        )
+
+    # MYPY: fix complains about overriding __ne__
+    def __ne__(self, other: Union[int, "BitVec"]) -> Bool:  # type: ignore
```

Comment:
Adding other operators like >=, <= might also be good, mypy caught some problems with the Bitvec class regarding usage of >= and <= in some part of code.It might also happen for this class

Reply:
I do not believe `functools.total_ordering()` will work because we are not returning a `bool`, but a custom class. I'll implement `<=` and `>=` though.